src/azure-cli/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -190,15 +205,83 @@ def _scaffold_certificate_profile(cmd):
     return template
 
 
-def list_keyvault(client, resource_group_name=None):
+def delete_vault_or_hsm(cmd, client, resource_group_name=None, vault_name=None, hsm_name=None):
```

Comment:
for each func below, I think it's necessary to check valut_name and hsm_name is mutual exclusive